### PR TITLE
Add an echo server test

### PR DESCRIPTION
**Pull request by justinj**
Still rough around the edges, connect_tcp in particular is not quite right, but
wanted to get this out there to unblock @jhellerstein.

### CODE REVIEW

File: hydroflow/src/scheduled/net/network_vertex.rs [hunk 1]
```diff
@@ -18,6 +21,100 @@ pub type ReceiverId = u32;
...
+    where
+        A: ToSocketAddrs,
+    {
+        let stream = TcpStream::connect(addr).await.unwrap();
+
+        let (reader, writer) = stream.into_split();
+
+        let (mut incoming_send, incoming_messages) = futures::channel::mpsc::channel(1024);
+        let mut reader = FramedRead::new(reader, LengthDelimitedCodec::new());
+        tokio::spawn(async move {
```
Comment (MingweiSamuel) [inline hydroflow/src/scheduled/net/network_vertex.rs:95]: TcpStream/FramedRead etc implement Stream, would somehow chaining directly like this instead work?
```rust
let my_stream = tcp_stream.map(|msg| Message::decode(&msg.freeze()).ok());
self.add_input_from_stream(my_stream);
```

File: hydroflow/src/scheduled/net/network_vertex.rs [hunk 2]
```diff
@@ -18,6 +21,100 @@ pub type ReceiverId = u32;
...
+        let port = listener.local_addr().unwrap().port();
+
+        let (mut conns, egress_edge) = self.add_egress_vertex();
+        // TODO(justin): figure out an appropriate buffer here.
+        let (incoming_send, incoming_messages) = futures::channel::mpsc::channel(1024);
+        let (mut incoming_conns, conns_recv) = futures::channel::mpsc::channel(1024);
+
+        // Listen to incoming connections and spawn a tokio task for each one,
+        // which feeds into the channel.
+        tokio::spawn(async move {
```
Comment (MingweiSamuel) [inline hydroflow/src/scheduled/net/network_vertex.rs:44]: Loose hidden `tokio::spawn` is suspicious but I'm fine with a TODO here for fixing it

**Review by MingweiSamuel — APPROVED** [review]
Just address comments then 👌

File: hydroflow/src/scheduled/net/network_vertex.rs (hunk 2 above)
Comment (justinj) [reply hydroflow/src/scheduled/net/network_vertex.rs:44]: What's suspicious about it? Don't we have to spawn to continually grab incoming connections? Do you mean we don't have a way to shutdown the server?

Comment (MingweiSamuel) [reply hydroflow/src/scheduled/net/network_vertex.rs:44]: Yeah, no way to shut down or handle errors, hidden side effects of calling a method

File: hydroflow/src/scheduled/net/network_vertex.rs (hunk 1 above)
Comment (justinj) [reply hydroflow/src/scheduled/net/network_vertex.rs:95]: nice, thats way better.

File: hydroflow/src/scheduled/net/network_vertex.rs (hunk 2 above)
Comment (jhellerstein) [reply hydroflow/src/scheduled/net/network_vertex.rs:45]: tokio::spawn returns a `JoinHandle`, so should be able to track. https://tokio.rs/tokio/tutorial/spawning

**Pull request merged**